### PR TITLE
*: implement other essential functions for raftkv2

### PR DESCRIPTION
**Pull request by BusyJay**
### What is changed and how it works?
Issue Number: Ref #12842 

What's Changed:

```commit-message
Functions like flashback will be supported in next milestone.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/raftstore-v2/src/batch/store.rs [hunk 1]
```diff
@@ -508,6 +508,14 @@ impl<EK: KvEngine, ER: RaftEngine> StoreRouter<EK, ER> {
         &self.logger
     }
 
+    #[inline]
+    pub fn check_send(&self, addr: u64, msg: PeerMsg) -> crate::Result<()> {
```
Comment (tabokie) [inline components/raftstore-v2/src/batch/store.rs:512]: What does the name mean?

File: components/raftstore/src/store/fsm/peer.rs [hunk 2]
```diff
@@ -5553,7 +5553,28 @@ where
             "split_keys" => %KeysInfoFormatter(split_keys.iter()),
             "source" => source,
         );
-        if let Err(e) = self.validate_split_region(&region_epoch, &split_keys) {
+
+        if !self.fsm.peer.is_leader() {
+            // region on this store is no longer leader, skipped.
+            info!(
+                "not leader, skip.";
```
Comment (tabokie) [inline components/raftstore/src/store/fsm/peer.rs:5560]: "prepare split skipped."

**Review by tabokie — APPROVED** [review]

File: components/raftstore-v2/src/batch/store.rs (hunk 1 above)
Comment (BusyJay) [reply components/raftstore-v2/src/batch/store.rs:512]: Send and check the result. I will convert `Disconnected` to `RegionNotFound` error.

File: components/raftstore/src/store/fsm/peer.rs [hunk 3]
```diff
@@ -5553,7 +5553,28 @@ where
...
+            return;
+        }
+        if let Err(e) = util::validate_split_region(
+            self.fsm.region_id(),
+            self.fsm.peer_id(),
+            self.region(),
+            &region_epoch,
+            &split_keys,
+        ) {
+            info!("invalid split request"; "err" => ?e, "region_id" => self.fsm.region_id(), "peer_id" => self.fsm.peer_id(), "source" => %source);
```
Comment (Connor1996) [inline components/raftstore/src/store/fsm/peer.rs:5577]: format?

**Review by Connor1996 — APPROVED** [review]
rest LGTM

**Pull request merged**